### PR TITLE
`azurerm_postgresql_flexible_server` - support for `authentication`, `azurerm_postgresql_flexible_server_active_directory_administrator` - add new resource

### LINKED ISSUE #18838: Support for Azure Active Directory for Azure Database for PostgreSQL – Flexible Server
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Community Note

* Please vote on this issue by adding a :thumbsup: [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Description

Azure Active Directory authentication for Azure Database for PostgreSQL - Flexible Server allows to improve database security by delegating credential management and authentication to a centralized identity provider. Azure Active Directory supports advanced security features such as second factor authentication options, password lifecycle management, applications and managed identities and conditional access. Azure Active Directory for Azure Database for PostgreSQL – Flexible Server now provides full support for managed identities, improved group roles and support for invited users and Azure Active Directory-only authentication mode with ability to disable local user support.

### New or Affected Resource(s)/Data Source(s)

azurerm_postgresql_flexible_server_active_directory_administrator

### Potential Terraform Configuration

```hcl
resource "azurerm_postgresql_flexible_server_active_directory_administrator" "example" {
  server_name         = azurerm_postgresql_server.example.name
  resource_group_name = azurerm_resource_group.example.name
  login               = "sqladmin"
  tenant_id           = data.azurerm_client_config.current.tenant_id
  object_id           = data.azurerm_client_config.current.object_id
}
```

### References

https://learn.microsoft.com/en-us/azure/postgresql/flexible-server/concepts-azure-ad-authentication

### PR DESCRIPTION
**Pull request by ziyeqf**
fix #18838

Test (run on my machine, the TC subscription has met the quota limitation)
---
```
TF_ACC=1 go test -v ./internal/services/postgres -run=TestAccPostgresqlFlexibleServer -timeout=600m
=== RUN   TestAccPostgresqlFlexibleServerAdministrator_basic
=== PAUSE TestAccPostgresqlFlexibleServerAdministrator_basic
=== RUN   TestAccPostgresqlFlexibleServerAdministrator_requiresImport
=== PAUSE TestAccPostgresqlFlexibleServerAdministrator_requiresImport
=== RUN   TestAccPostgresqlFlexibleServerAdministrator_disappears
=== PAUSE TestAccPostgresqlFlexibleServerAdministrator_disappears
=== RUN   TestAccPostgresqlFlexibleServerDatabase_basic
=== PAUSE TestAccPostgresqlFlexibleServerDatabase_basic
=== RUN   TestAccPostgresqlFlexibleServerDatabase_requiresImport
=== PAUSE TestAccPostgresqlFlexibleServerDatabase_requiresImport
=== RUN   TestAccPostgresqlFlexibleServerDatabase_charsetLowercase
=== PAUSE TestAccPostgresqlFlexibleServerDatabase_charsetLowercase
=== RUN   TestAccPostgresqlFlexibleServerDatabase_withoutCharsetAndCollation
=== PAUSE TestAccPostgresqlFlexibleServerDatabase_withoutCharsetAndCollation
=== RUN   TestAccPostgresqlFlexibleServerFirewallRule_basic
=== PAUSE TestAccPostgresqlFlexibleServerFirewallRule_basic
=== RUN   TestAccPostgresqlFlexibleServerFirewallRule_requiresImport
=== PAUSE TestAccPostgresqlFlexibleServerFirewallRule_requiresImport
=== RUN   TestAccPostgresqlFlexibleServerFirewallRule_update
=== PAUSE TestAccPostgresqlFlexibleServerFirewallRule_update
=== RUN   TestAccPostgresqlFlexibleServer_basic
=== PAUSE TestAccPostgresqlFlexibleServer_basic
=== RUN   TestAccPostgresqlFlexibleServer_requiresImport
=== PAUSE TestAccPostgresqlFlexibleServer_requiresImport
=== RUN   TestAccPostgresqlFlexibleServer_complete
=== PAUSE TestAccPostgresqlFlexibleServer_complete
=== RUN   TestAccPostgresqlFlexibleServer_completeUpdate
=== PAUSE TestAccPostgresqlFlexibleServer_completeUpdate
=== RUN   TestAccPostgresqlFlexibleServer_updateMaintenanceWindow
=== PAUSE TestAccPostgresqlFlexibleServer_updateMaintenanceWindow
=== RUN   TestAccPostgresqlFlexibleServer_updateSku
=== PAUSE TestAccPostgresqlFlexibleServer_updateSku
=== RUN   TestAccPostgresqlFlexibleServer_pointInTimeRestore
=== PAUSE TestAccPostgresqlFlexibleServer_pointInTimeRestore
=== RUN   TestAccPostgresqlFlexibleServer_failover
=== PAUSE TestAccPostgresqlFlexibleServer_failover
=== RUN   TestAccPostgresqlFlexibleServer_geoRedundantBackupEnabled
=== PAUSE TestAccPostgresqlFlexibleServer_geoRedundantBackupEnabled
=== RUN   TestAccPostgresqlFlexibleServer_authConfig
=== PAUSE TestAccPostgresqlFlexibleServer_authConfig
=== CONT  TestAccPostgresqlFlexibleServerAdministrator_basic
=== CONT  TestAccPostgresqlFlexibleServer_basic
=== CONT  TestAccPostgresqlFlexibleServerDatabase_charsetLowercase
=== CONT  TestAccPostgresqlFlexibleServer_updateSku
=== CONT  TestAccPostgresqlFlexibleServerFirewallRule_requiresImport
=== CONT  TestAccPostgresqlFlexibleServer_geoRedundantBackupEnabled
=== CONT  TestAccPostgresqlFlexibleServerFirewallRule_basic
=== CONT  TestAccPostgresqlFlexibleServerDatabase_withoutCharsetAndCollation
--- PASS: TestAccPostgresqlFlexibleServer_basic (807.88s)
=== CONT  TestAccPostgresqlFlexibleServer_failover
--- PASS: TestAccPostgresqlFlexibleServer_geoRedundantBackupEnabled (811.99s)
=== CONT  TestAccPostgresqlFlexibleServer_pointInTimeRestore
--- PASS: TestAccPostgresqlFlexibleServerDatabase_withoutCharsetAndCollation (835.71s)
=== CONT  TestAccPostgresqlFlexibleServerFirewallRule_update
--- PASS: TestAccPostgresqlFlexibleServerDatabase_charsetLowercase (841.78s)
=== CONT  TestAccPostgresqlFlexibleServer_authConfig
--- PASS: TestAccPostgresqlFlexibleServerFirewallRule_basic (933.95s)
=== CONT  TestAccPostgresqlFlexibleServerDatabase_basic
--- PASS: TestAccPostgresqlFlexibleServerAdministrator_basic (1001.35s)
=== CONT  TestAccPostgresqlFlexibleServerDatabase_requiresImport
--- PASS: TestAccPostgresqlFlexibleServerFirewallRule_requiresImport (1014.78s)
=== CONT  TestAccPostgresqlFlexibleServerAdministrator_disappears
--- PASS: TestAccPostgresqlFlexibleServerDatabase_basic (898.10s)
=== CONT  TestAccPostgresqlFlexibleServer_completeUpdate
--- PASS: TestAccPostgresqlFlexibleServerDatabase_requiresImport (863.23s)
=== CONT  TestAccPostgresqlFlexibleServer_updateMaintenanceWindow
--- PASS: TestAccPostgresqlFlexibleServerAdministrator_disappears (885.03s)
=== CONT  TestAccPostgresqlFlexibleServer_complete
--- PASS: TestAccPostgresqlFlexibleServer_updateSku (1921.77s)
=== CONT  TestAccPostgresqlFlexibleServer_requiresImport
--- PASS: TestAccPostgresqlFlexibleServer_authConfig (1226.91s)
=== CONT  TestAccPostgresqlFlexibleServerAdministrator_requiresImport
--- PASS: TestAccPostgresqlFlexibleServerFirewallRule_update (1309.66s)
=== CONT  TestAccPostgresqlFlexibleServer_pointInTimeRestore
    testcase.go:110: Step 3/4 error: Error running apply: exit status 1

        Error: creating Flexible Server (Subscription: "85b3dbca-5974-4067-9669-67a141095a76"
        Resource Group Name: "acctestRG-postgresql-221115112156141913"
        Server Name: "acctest-fs-pitr-221115112156141913"): polling after Create: Code="InternalServerError" Message="An unexpected error occured while processing the request. Tracking ID: '549865cc-4c98-4008-9727-f1b3bc6c49fe'"

          with azurerm_postgresql_flexible_server.pitr,
          on terraform_plugin_test.tf line 31, in resource "azurerm_postgresql_flexible_server" "pitr":
          31: resource "azurerm_postgresql_flexible_server" "pitr" {

--- FAIL: TestAccPostgresqlFlexibleServer_pointInTimeRestore (1976.75s)
--- PASS: TestAccPostgresqlFlexibleServer_requiresImport (942.95s)
--- PASS: TestAccPostgresqlFlexibleServer_updateMaintenanceWindow (1274.01s)
--- PASS: TestAccPostgresqlFlexibleServerAdministrator_requiresImport (1080.43s)
--- PASS: TestAccPostgresqlFlexibleServer_complete (1467.55s)
--- PASS: TestAccPostgresqlFlexibleServer_failover (2769.25s)
--- PASS: TestAccPostgresqlFlexibleServer_completeUpdate (3054.62s)
FAIL
FAIL    github.com/hashicorp/terraform-provider-azurerm/internal/services/postgres      4886.688s
```

`TestAccPostgresqlFlexibleServer_pointInTimeRestore` failed due to server internal error, so re runed it.
```
TF_ACC=1 go test -v ./internal/services/postgres -run=TestAccPostgresqlFlexibleServer_pointInTimeRestore -timeout=60m 
=== RUN   TestAccPostgresqlFlexibleServer_pointInTimeRestore
=== PAUSE TestAccPostgresqlFlexibleServer_pointInTimeRestore
=== CONT  TestAccPostgresqlFlexibleServer_pointInTimeRestore
--- PASS: TestAccPostgresqlFlexibleServer_pointInTimeRestore (2514.04s)
PASS
ok      github.com/hashicorp/terraform-provider-azurerm/internal/services/postgres      2515.213s
```

### CODE REVIEW

**Review by katbyte — CHANGES_REQUESTED** [review]
noticed while passing through

File: website/docs/r/postgresql_flexible_server_active_directory_administrator.html.markdown [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+
+* `resource_group_name` - (Required) The name of the resource group for the PostgreSQL server. Changing this forces a new resource to be created.
+
+* `object_id` - (Required) The object ID of a user, service principal or security group in the Azure Active Directory tenant set as the Flexible Server Admin.
+
+* `tenant_id` - (Required) The Azure Tenant ID.
+
+* `principal_name` - (Required) The name of Azure active directory principal.
+
+* `principal_type` - (Required) The type of Azure active directory principal. Possible values are `Group`, `ServicePrincipal` and `User`.
```
Comment (catriona-m) [inline website/docs/r/postgresql_flexible_server_active_directory_administrator.html.markdown:63]: It looks like these are ForceNew in the schema, so we should add Changing this forces a new resource to be created. to these descriptions.

**Review by catriona-m — CHANGES_REQUESTED** [review]
Thanks @ziyeqf, I noticed a couple of other minor things that could be fixed up when looking through again, but overall this is looking good. Thanks!

**Review by catriona-m — APPROVED** [review]
Thanks @ziyeqf LGTM!

**Pull request merged**